packages/manager/apps/pci-kubernetes/src/components/create/NetworkClusterStep.component.tsx

```diff
@@ -0,0 +1,133 @@
+import { ODS_SPINNER_SIZE } from '@ovhcloud/ods-components';
+import { OsdsSpinner } from '@ovhcloud/ods-components/react';
+import { useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useParams } from 'react-router-dom';
+import { usePrivateNetworkSubnets } from '@/api/hooks/useSubnets';
+import { useAvailablePrivateNetworks } from '@/api/hooks/useNetwork';
+import LoadBalancerSelect from './LoadBalancerSelect.component';
+import PrivateNetworkSelect from './PrivateNetworkSelect.component';
+import PublicPrivateGateway from './PublicPrivateGateway.component';
+import SubnetSelect from './SubnetSelect.component';
+import { TNetwork } from '@/api/data/network';
+import { TPrivateNetworkSubnet } from '@/api/data/subnets';
+
+export type TFormState = {
+  regionName: string;
+  gatewayIp: string;
+  gatewayEnabled: boolean;
+  privateNetwork: TNetwork;
+  subnet: TPrivateNetworkSubnet;
+  loadBalancersSubnet: TPrivateNetworkSubnet;
+};
+
+export type NetworkClusterStepProps = {
+  formState: TFormState;
+  setFormState: (formState) => void;
+};
+
+export default function NetworkClusterStep({
+  formState,
+  setFormState,
+}: Readonly<NetworkClusterStepProps>) {
+  const { t } = useTranslation('network-add');
+  const { projectId } = useParams();
+
+  const defaultPrivateNetwork = {
+    id: null,
+    name: t('kubernetes_network_form_none'),
+  } as TNetwork;
+
+  const {
+    data: availablePrivateNetworks,
+    isPending,
+  } = useAvailablePrivateNetworks(projectId, formState.regionName);
+
+  const {
+    subnets,
+    subnetsByRegion,
+    isLoading: isSubnetsLoading,
+    error: subnetError,
+  } = usePrivateNetworkSubnets(
+    projectId,
+    formState.privateNetwork?.id,
+    formState.regionName,
+  );
+
+  const hasPrivateNetwork = Boolean(formState.privateNetwork?.id);
+
+  const isSubnetShown =
+    hasPrivateNetwork && !isSubnetsLoading && !!subnetsByRegion?.length;
+
+  const isGatewayShown = hasPrivateNetwork && !isSubnetsLoading;
+
+  useEffect(() => {
+    setFormState((prev) => ({
+      ...prev,
+      privateNetwork: defaultPrivateNetwork,
+    }));
+  }, []);
```

Comment:
i'm not sure to understand this part, why can we provide privateNetwork as props if it's always overridden by defaultPrivateNetwork ?

also isn't hasPrivateNetwork always true since you are settings the value of private network to defaultprivatenetwork ?